pandas/core/groupby.py

```diff
@@ -2613,6 +2613,13 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
 
             level = None
             key = group_axis
+        elif key is None:
```

Comment:
hate adding logic here, this function is already impenetrable, can you incorporate this to existing?